### PR TITLE
add base configuration setting for commnuication with programs, organ…

### PR DESCRIPTION
**Pull request by zubair-arbi**
…ization and user api

ECOM-3455
@awais786 @ahsan-ul-haq @tasawernawaz

### CODE REVIEW

**Comment by zubair-arbi** [comment]
@clintonb @jimabramson @feanil please review

**Comment by zubair-arbi** [comment]
fyi: @e0d @arbabnazar

**Comment by zubair-arbi** [comment]
Here is the console log for my sandbox created with these settings: http://jenkins.edx.org:8080/view/Ansible/job/ansible-provision/5052/console

File: playbooks/roles/credentials/defaults/main.yml [hunk 1]
```diff
@@ -65,6 +65,27 @@ CREDENTIALS_SOCIAL_AUTH_REDIRECT_IS_HTTPS: false
 
 CREDENTIALS_PLATFORM_NAME: 'Your Platform Name Here'
 
+# credentials service user in programs service and lms
+CREDENTIALS_SERVICE_USER: 'credentials_service_user'
+
+# Absolute URL used to get programs from the programs service.
+PROGRAMS_API_URL: 'https://127.0.0.1:8004/api/v1/'
```
Comment (clintonb) [inline playbooks/roles/credentials/defaults/main.yml]: These should all be prefixed with `CREDENTIALS_`.

File: playbooks/roles/credentials/defaults/main.yml [hunk 2]
```diff
@@ -65,6 +65,27 @@ CREDENTIALS_SOCIAL_AUTH_REDIRECT_IS_HTTPS: false
...
+# JWT key and secret for programs service.
```
Comment (clintonb) [inline playbooks/roles/credentials/defaults/main.yml]: Remove. It's obvious from the variable names what these values are.

File: playbooks/roles/credentials/defaults/main.yml [hunk 3]
```diff
@@ -65,6 +65,27 @@ CREDENTIALS_SOCIAL_AUTH_REDIRECT_IS_HTTPS: false
...
+PROGRAMS_JWT_AUDIENCE: 'programs-key'
+PROGRAMS_JWT_SECRET_KEY: 'programs-secret'
+
+# Absolute URL used to get organization data from the organizations api in LMS
+ORGANIZATIONS_API_URL: 'https://127.0.0.1:8000/api/organizations/v0/'
+# JWT key and secret for organizations api.
```
Comment (clintonb) [inline playbooks/roles/credentials/defaults/main.yml]: Remove. It's obvious from the variable names what these values are.

File: playbooks/roles/credentials/defaults/main.yml [hunk 4]
```diff
@@ -65,6 +65,27 @@ CREDENTIALS_SOCIAL_AUTH_REDIRECT_IS_HTTPS: false
...
+ORGANIZATIONS_AUDIENCE: 'lms-key'
+ORGANIZATIONS_SECRET_KEY: 'lms-secret'
+
+# Absolute URL used to get user data from the user api in LMS
+USER_API_URL: 'https://127.0.0.1:8000/api/user/v1/'
+# JWT key and secret for user api.
```
Comment (clintonb) [inline playbooks/roles/credentials/defaults/main.yml]: Remove. It's obvious from the variable names what these values are.

File: playbooks/roles/credentials/defaults/main.yml [hunk 5]
```diff
@@ -65,6 +65,27 @@ CREDENTIALS_SOCIAL_AUTH_REDIRECT_IS_HTTPS: false
...
+# Absolute URL used to get programs from the programs service.
+PROGRAMS_API_URL: 'https://127.0.0.1:8004/api/v1/'
+# JWT key and secret for programs service.
+PROGRAMS_JWT_AUDIENCE: 'programs-key'
+PROGRAMS_JWT_SECRET_KEY: 'programs-secret'
+
+# Absolute URL used to get organization data from the organizations api in LMS
+ORGANIZATIONS_API_URL: 'https://127.0.0.1:8000/api/organizations/v0/'
+# JWT key and secret for organizations api.
+ORGANIZATIONS_AUDIENCE: 'lms-key'
```
Comment (clintonb) [inline playbooks/roles/credentials/defaults/main.yml]: `ORGNIZATIONS_API_*`

File: playbooks/roles/credentials/defaults/main.yml [hunk 6]
```diff
@@ -65,6 +65,27 @@ CREDENTIALS_SOCIAL_AUTH_REDIRECT_IS_HTTPS: false
 
 CREDENTIALS_PLATFORM_NAME: 'Your Platform Name Here'
 
+# credentials service user in programs service and lms
+CREDENTIALS_SERVICE_USER: 'credentials_service_user'
+
+# Absolute URL used to get programs from the programs service.
+PROGRAMS_API_URL: 'https://127.0.0.1:8004/api/v1/'
+# JWT key and secret for programs service.
+PROGRAMS_JWT_AUDIENCE: 'programs-key'
```
Comment (clintonb) [inline playbooks/roles/credentials/defaults/main.yml]: `PROGRAMS_API_*`

File: playbooks/roles/credentials/defaults/main.yml [hunk 7]
```diff
@@ -65,6 +65,27 @@ CREDENTIALS_SOCIAL_AUTH_REDIRECT_IS_HTTPS: false
...
+# Absolute URL used to get organization data from the organizations api in LMS
+ORGANIZATIONS_API_URL: 'https://127.0.0.1:8000/api/organizations/v0/'
+# JWT key and secret for organizations api.
+ORGANIZATIONS_AUDIENCE: 'lms-key'
+ORGANIZATIONS_SECRET_KEY: 'lms-secret'
+
+# Absolute URL used to get user data from the user api in LMS
+USER_API_URL: 'https://127.0.0.1:8000/api/user/v1/'
+# JWT key and secret for user api.
+USER_JWT_AUDIENCE: 'lms-key'
```
Comment (clintonb) [inline playbooks/roles/credentials/defaults/main.yml]: `USER_API_*`

**Comment by jimabramson** [comment]
@peter-fogg could you take my spot on this review?

**Comment by zubair-arbi** [comment]
@clintonb I have addressed your feedback please review

**Comment by clintonb** [comment]
:+1:

**Comment by peter-fogg** [comment]
LGTM, though tests are failing and I can't quite tell what's going on from the Travis logs.

**Comment by zubair-arbi** [comment]
@e0d @feanil this PR is failing again and again on this task:

```
TASK: [edxapp | install python requirements] ********************************** 
```

Here are previous two failures:
1. https://travis-ci.org/edx/configuration/jobs/102562230
2. https://travis-ci.org/edx/configuration/jobs/102572681

**Comment by zubair-arbi** [comment]
@arbabnazar  can you please look into this issue?

**Comment by zubair-arbi** [comment]
@fredsmith @e0d the build failed again: https://travis-ci.org/edx/configuration/jobs/102572681

**Comment by e0d** [comment]
@cpennington why is this change running the edxapp play?

**Comment by cpennington** [comment]
Because it was rebased on master, where you enabled edxapp.

-Cale

**Comment by zubair-arbi** [comment]
@e0d @fredsmith any suggestion on how to resolve the failing build?

**Comment by cpennington** [comment]
@zubair-arbi you could try changing the `travis_wait 30` to `travis_wait 40`. Longer term, we need to make it so that travis is seeing incremental progress during that part of the edxapp build.

**Comment by zubair-arbi** [comment]
@cpennington @e0d the build pass with increase `travis` wait time. Should I merge it?

**Comment by e0d** [comment]
:+1:

**Comment by cpennington** [comment]
:+1:

**Pull request merged**